script/clang-format-check.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+res=0
+
+clang_format_binary="$1"
+
+shift
+
+for v in $@; do
+    comp_res=$("$clang_format_binary" -style=file "$v" | diff "$v" -)
+    if [ ! -z "$comp_res" ] ; then
+        echo "$v failed the clang-format conformance check." >&2
```

Comment:
please output `$comp_res` else it is not clear what is wrong

Reply:
What do you get from looking at the return code when you already know that it was nonzero?

Reply:
This variable contains the diff and show you all the wrong formated lines.

Reply:
I added printing the variable content